setup.py

```diff
@@ -255,31 +250,52 @@
     'pufferlib/ocean/tripletriad/cy_tripletriad',
     'pufferlib/ocean/go/cy_go',
     'pufferlib/ocean/rware/cy_rware',
-    'pufferlib/ocean/trash_pickup/cy_trash_pickup'
+    'pufferlib/ocean/trash_pickup/cy_trash_pickup',
 ]
 
-extensions = [Extension(
-    path.replace('/', '.'),
-    [path + '.pyx'],
-    include_dirs=[numpy.get_include(), 'raylib/include'],
-    library_dirs=['raylib/lib'],
-    libraries=["raylib"],
-    runtime_library_dirs=["raylib/lib"],
-    extra_compile_args=['-DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION', '-DPLATFORM_DESKTOP', '-O2', '-Wno-alloc-size-larger-than'],#, '-g'],
-    extra_link_args=["-Wl,-rpath,$ORIGIN/raylib/lib"]
+system = platform.system()
+
+if system == 'Darwin':
+    # On macOS, use @loader_path.
+    # The extension “.so” is typically in pufferlib/ocean/...,
+    # and “raylib/lib” is (maybe) two directories up from ocean/<env>.
+    # So @loader_path/../../raylib/lib is common.
+    rpath_arg = '-Wl,-rpath,@loader_path/../../raylib/lib'
```

Comment:
I ran into a variety of linkage issues here, this was the best I could come up with. Happy to hear any better ideas or solutions to make this cleaner across platforms.